fluent-plugin-events/lib/fluent/plugin/in_events.rb

```diff
@@ -43,90 +44,139 @@ def configure(conf)
+          if now - last_recreated >= @watch_interval_seconds || !watcher_exists
+            
+            log.debug "Recreating watcher thread. Use resource version from latest snapshot if watcher is running"
+            pull_resource_version if watcher_exists
+            @watch_stream.finish if @watch_stream
+
+            start_watcher_thread
+            last_recreated = now
+            log.debug "last_recreated updated to #{last_recreated}"
+          end
+
+          update_config_map
+          sleep(@configmap_update_interval_seconds)
         end
       end
   
       def start_watcher_thread
+        log.debug "Starting watcher thread"
         params = Hash.new
         params[:as] = :raw
         params[:resource_version] = @resource_version
         params[:field_selector] = @field_selector
         params[:label_selector] = @label_selector
         params[:namespace] = @namespace
+        params[:timeout_seconds] = @watch_interval_seconds + 60
 
         @watcher = @client.public_send("watch_events", params).tap do |watcher|
           thread_create(:"watch_events") do
-            watcher.each do |entity|
-              log.debug "Received new object from watching events"
+            @watch_stream = watcher
+            @watcher_id = Thread.current.object_id
+            log.debug "New thread to watch events #{@watcher_id} from resource version #{params[:resource_version]}"
 
+            watcher.each do |entity|
               begin
                 entity = JSON.parse(entity)
                 router.emit tag, Fluent::Engine.now, entity
-                @resource_version = entity['object']['metadata']['resourceVersion']
+                rv = entity['object']['metadata']['resourceVersion']
               rescue => e
                 log.error "Got exception #{e} parsing entity #{entity}. Skipping."
               end
 
-              if (!@resource_version)
-                @resource_version = 0
-                sleep(5)
-                start_watcher_thread
+              if (!rv)
+                log.error "Resource version #{rv} expired, waiting for stream to be recreated with more recent version."
                 break
               end
             end
+
+            log.debug "Closing watch stream"
           end
         end
       end
 
-      def start_configmap_flush_thread
-        thread_create(:"update_configmap") do
-          resource = ::Kubeclient::Resource.new
-          resource.metadata = {
-            name: "fluentd-config-resource-version",
-            namespace: "sumologic"
-          }
-          while true do
-            sleep(@configmap_update_interval_seconds)
-            resource.data = { "resource-version": "#{@resource_version}"}
-
-            # update the config map
-            @client.public_send("update_config_map", resource).tap do |maps|
-              log.debug "Updated config maps: #{maps}"
-            end
+      def create_config_maps
```

Comment:
oops, sorry forgot to add the nit comment: since we have only one config map, can we change all the `maps` to `map`?